src/Mailgun/Mailgun.php

```diff
@@ -84,20 +84,20 @@ public function __construct(
     }
 
     /**
-     * @param HttpClientConfigurator $httpClientConfigurator
+     * @param HttpClientConfigurator $configurator
      * @param Hydrator|null          $hydrator
      * @param RequestBuilder|null    $requestBuilder
      *
      * @return Mailgun
      */
     public static function configure(
-        HttpClientConfigurator $httpClientConfigurator,
+        HttpClientConfigurator $configurator,
         Hydrator $hydrator = null,
         RequestBuilder $requestBuilder = null
     ) {
-        $httpClient = $httpClientConfigurator->createConfiguredClient();
+        $httpClient = $configurator->createConfiguredClient();
 
-        return new self(null, $httpClient, 'api.mailgun.net', $hydrator, $requestBuilder);
+        return new self($configurator->getApiKey(), $httpClient, 'api.mailgun.net', $hydrator, $requestBuilder);
```

Comment:
Why this change? It was not here until now, and I don't see calls to this static method (or I am too tired).